transform-plugins/src/test/java/co/cask/hydrator/plugin/ValueMapperTest.java

```diff
@@ -0,0 +1,715 @@
+    dataSetManager.flush();
+
+    addDatasetInstance(KeyValueTable.class.getName(), "salary_lookup_table");
+    DataSetManager<KeyValueTable> salaryDataSetManager = getDataset("salary_lookup_table");
+    KeyValueTable dsalaryKeyValueTable = salaryDataSetManager.get();
+    dsalaryKeyValueTable.write("1000".getBytes(Charsets.UTF_8), "Low".getBytes(Charsets.UTF_8));
+    dsalaryKeyValueTable.write("2000".getBytes(Charsets.UTF_8), "Medium".getBytes(Charsets.UTF_8));
+    dsalaryKeyValueTable.write("5000".getBytes(Charsets.UTF_8), "High".getBytes(Charsets.UTF_8));
+    salaryDataSetManager.flush();
+
+    DataSetManager<Table> inputManager = getDataset(inputTable);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "100").set(NAME, "John").set(SALARY, "1000")
+        .set(DESIGNATIONID, "1").build(),
+      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "101").set(NAME, "Kerry").set(SALARY, "2000")
+        .set(DESIGNATIONID, "2").build(),
+      StructuredRecord.builder(SOURCE_SCHEMA).set(ID, "102").set(NAME, "Mathew").set(SALARY, "5000")
+        .set(DESIGNATIONID, "3").build()
+    );
+    MockSource.writeInput(inputManager, input);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(sinkTable);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
+
+    Map<String, String> nameDesignationMap = new HashMap<String, String>();
+    nameDesignationMap.put("John", "SE");
+    nameDesignationMap.put("Kerry", "SSE");
+    nameDesignationMap.put("Mathew", "ML");
+
+    StructuredRecord recordOne = outputRecords.get(0);
+    StructuredRecord recordTwo = outputRecords.get(1);
+    StructuredRecord recordThree = outputRecords.get(2);
+
+    Assert.assertEquals(3, outputRecords.size());
+    Assert.assertEquals(nameDesignationMap.get(recordOne.get(NAME)), outputRecords.get(0)
+      .get(DESIGNATIONNAME));
+    Assert.assertEquals(nameDesignationMap.get(recordTwo.get(NAME)), outputRecords.get(1)
+      .get(DESIGNATIONNAME));
+    Assert.assertEquals(nameDesignationMap.get(recordThree.get(NAME)), outputRecords.get(2)
+      .get(DESIGNATIONNAME));
+
+    Map<String, String> nameSalaryMap = new HashMap<String, String>();
+    nameSalaryMap.put("John", "Low");
+    nameSalaryMap.put("Kerry", "Medium");
+    nameSalaryMap.put("Mathew", "High");
+
+
+    Assert.assertEquals(nameSalaryMap.get(recordOne.get(NAME)), outputRecords.get(0)
+      .get(SALARYDESC));
+    Assert.assertEquals(nameSalaryMap.get(recordTwo.get(NAME)), outputRecords.get(1)
+      .get(SALARYDESC));
+    Assert.assertEquals(nameSalaryMap.get(recordThree.get(NAME)), outputRecords.get(2)
+      .get(SALARYDESC));
+
+  }
+
+  @Test
+  public void testTargetSchemaValidation() throws Exception {
```

Comment:
you don't need to create a pipeline to test schema validation.  You can just create the Transform directly and call configurePipeline with a mock configurer.  You could also test the output schema by doing that.